### PR TITLE
Added GitHub Actions module (gha)

### PR DESCRIPTION
**Pull request by applejag**
First iteration of GH module. Just a lot of very basic tools.

Highly inspired by the https://github.com/actions/toolkit library.

Question: Should these functions be renamed to `gha` module instead? As these are more GitHub Action related than just plain GitHub API functions? Or is it OK with having "get repository" and "create pullrequest" type of functions (planned) in the same module as these `gh.is_debug` and `gh.add_path`-type of functions?

Relates to #124 (but does not resolve it, there's still a lot of stuff missing)

### CODE REVIEW

File: modules/gh/gh.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+}
+
+func Module() *object.Module {
+	return object.NewBuiltinsModule("gh", map[string]object.Object{
+		"is_debug":    object.NewBuiltin("gh.is_debug", IsDebug),
+		"log_info":    object.NewBuiltin("gh.log_info", LogInfo),
+		"log_debug":   object.NewBuiltin("gh.log_debug", LogDebug),
+		"log_notice":  object.NewBuiltin("gh.log_notice", LogNotice),
+		"log_warning": object.NewBuiltin("gh.log_warning", LogWarning),
+		"log_error":   object.NewBuiltin("gh.log_error", LogError),
```
Comment (luisdavim) [inline modules/gh/gh.go:297]: Are all these needed? We can check env vars and print messages with `risor` already....

Comment (applejag) [reply modules/gh/gh.go:297]: The logging ones are helper functions, as handling multiline messages and the file parameters can be quite tedious when they're based on variables.

The `log_info` I could agree we can remove, as it's practically just an alias for `println`. 

Regarding `is_debug` it exposes the API in a nicer way. But I'm OK with removing it too, if you think it bloats the API.

The remaining logger functions I really think should stay though.

**Comment by luisdavim** [comment]
can you add an example under `examples/scripts` I think all new features should come with one of those, it helps reviewing and they also serve as documentation, and latter could actually be included in the documentation...

File: modules/gh/gh.go (hunk 1 above)
Comment (applejag) [reply modules/gh/gh.go]: I've removed `log_info`

Comment (applejag) [reply modules/gh/gh.go]: Oh and I forgot to mention, these functions were practically ported from here: https://github.com/actions/toolkit/blob/1fe633e27c4cc74616d675b71163f59f9d084381/packages/core/src/core.ts#L297-L299

That's why I added the `log_info` to begin with.

File: modules/gh/gh.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+func appendWorkflowFile(path string, message string) object.Object {
+	// Using the Go "os" package instead of the Risor "os" package
+	// because the Risor application might use S3 as storage, but
+	// the GitHub Actions special files are still on the real OS' file system.
+	file, err := goos.OpenFile(path, goos.O_APPEND|goos.O_WRONLY|goos.O_CREATE, 0644)
+	if err != nil {
+		return object.Errorf("io error: %v", err)
+	}
```
Comment (myzie) [inline modules/gh/gh.go:117]: While it's true that Risor _could_ be configured to use s3 storage for the entire filesystem, I don't think that would be likely in a situation involving Github actions. Let's still use Risor's GetDefaultOS mechanism here. Direct OS access circumvents using Risor in sandboxing like situations.

**Review by myzie — COMMENTED** [review]
Thanks @applejag. See my couple suggestions.

File: modules/all/all.go [hunk 3]
```diff
@@ -27,6 +28,7 @@ func Builtins() map[string]object.Object {
 		"exec":     modExec.Module(),
 		"filepath": modFilepath.Module(),
 		"fmt":      modFmt.Module(),
+		"gh":       modGh.Module(),
```
[the remark this answers is not in the document]
Comment (applejag) [reply modules/all/all.go:31]: Renamed.

File: modules/gh/gh.go [hunk 4]
```diff
@@ -0,0 +1,291 @@
+//go:build gh
+// +build gh
```
[the remark this answers is not in the document]
Comment (applejag) [reply modules/gh/gh.go]: Removed.

File: modules/gh/gh.go (hunk 2 above)
Comment (applejag) [reply modules/gh/gh.go]: Alright. I've added a `OS.Append()` function to the OS interface so I could completely rely on the Risor OS

File: os/os.go [hunk 5]
```diff
@@ -33,6 +33,7 @@ type File interface {
 }
 
 type FS interface {
+	Append(name string) (File, error)
```
Comment (myzie) [inline os/os.go:36]: Would you mind switching this to be a more general `OpenFile` instead? My aim with this interface is to match the Go `os` and/or `io/fs` package functions.

So specifically the signature should be:

```
OpenFile(name string, flag int, perm FileMode) (File, error)
```

Supporting `OpenFile` with all the different flags on `s3fs` may be a project onto itself, so I'd suggest just having that implementation return a "not implemented" error for now. (I don't want to make the job too hard :-) 

The PathSeparator and PathListSeparator additions are good as-is.

Comment (applejag) [reply os/os.go]: Done

**Pull request merged**